source/common/router/upstream_request.cc

```diff
@@ -133,8 +133,9 @@ UpstreamRequest::UpstreamRequest(RouterFilterInterface& parent,
 
   // Set up the upstream HTTP filter manager.
   filter_manager_callbacks_ = std::make_unique<UpstreamRequestFilterManagerCallbacks>(*this);
+  // TODO(alyssawilk): Verify that bypassing virtual dispatch here was intentional
```

Comment:
cc @alyssawilk

Reply:
this should be fine - it's a virtual API but no subclasses expected.
I'd suggest adding a "final" in the class and commenting why.

Reply:
updated